modules/delegates/api/fetchDelegates.ts

```diff
@@ -302,3 +313,139 @@ export async function fetchDelegates(
 
   return delegatesResponse;
 }
+
+export async function fetchDelegatesPaginated(
+  queryParams: DelegatesAPIQueryParams
+): Promise</*DelegatesAPIResponse*/ any> {
+  const network = queryParams.network;
+  const { data: githubDelegates } = await fetchGithubDelegates(network);
+  const tags = getDelegateTags();
+  const ghVoteDelegateAddresses =
+    githubDelegates?.map(delegate => delegate.voteDelegateAddress.toLowerCase()) || null;
+  const migratedDelegateAddresses = (
+    [
+      ...Object.keys(delegateAddressLinks[network]),
+      ...Object.values(delegateAddressLinks[network])
+    ] as string[]
+  ).map(delegate => delegate.toLowerCase());
+
+  const chainId = networkNameToChainId(network);
+
+  const delegatesQueryFilter =
+    queryParams.status === 'RECOGNIZED'
+      ? {
+          or: [
+            { voteDelegate: { in: ghVoteDelegateAddresses } },
+            { delegate: { in: migratedDelegateAddresses } }
+          ]
+        }
+      : queryParams.status === 'SHADOW'
+      ? {
+          and: [
+            { voteDelegate: { notIn: ghVoteDelegateAddresses } },
+            { delegate: { notIn: migratedDelegateAddresses } }
+          ]
+        }
+      : null;
+
+  const delegatesQueryVariables = { ...queryParams };
+  if (delegatesQueryFilter) delegatesQueryVariables['filter'] = delegatesQueryFilter;
```

Comment:
do not use inline ifs

Reply:
I'll change these, I think I've used them in a couple of places